app/frontend/wallet/wallet.ts

```diff
@@ -0,0 +1,118 @@
+import {Account} from './account'
+import BlockchainExplorer from './blockchain-explorer'
+
+const Wallet = ({config, cryptoProvider}) => {
+  const blockchainExplorer = BlockchainExplorer(config)
+
+  const accounts: Array<ReturnType<typeof Account>> = []
+
+  function loadAccount(accountIndex: number) {
+    const newAccount = Account({
+      config,
+      cryptoProvider,
+      blockchainExplorer,
+      accountIndex,
+    })
+    accounts.push(newAccount)
```

Comment:
this can easily result in accidentally loading the same account repeatedly and it also heavily relies on the accounts being loaded in the right order - either `accounts` should be an object and deduplication/ordering should be ensured by writing to the keys, or there should be a safeguard that would remove old instances of the same account from the list and sort them accordingly - I find it a bit more fool-proof to just use an object